protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -26,7 +30,35 @@ open class CodeGenerator(
         fn().also { indent = indent.dropLast(4) }
     }
 
-    protected fun writeType(type: File.Type, parentType: String? = null) = when (type) {
+    private fun writeExtension(field: File.Field) {
+        when (field) {
+            is File.Field.Numbered -> {
+                val localType = when (field.type) {
+                    File.Field.Type.MESSAGE -> ((field as File.Field.Numbered.Standard).localTypeName ?: field.kotlinLocalTypeName)?.substringAfter(".")
+                    else -> field.type.standardTypeName
+                }
+                val fieldExtendee = field.extendee!!.trimStart('.').replace("google.protobuf", "pbandk.wkt")
+                line(
+                    "val $fieldExtendee.${field.name}: ${localType}? get() ="
+                )
+                line("  getExtension(${file.packageName}.${field.name})")
+                line()
+                line("val ${field.name} = pbandk.FieldDescriptor(").indented {
+                    line("name = \"${field.name}\",")
+                    line("number = ${field.number},")
+                    line("type = ${field.fieldDescriptorType(false)},")
+                    line("value = $fieldExtendee::${field.name},")
+                    line("messageDescriptor = $fieldExtendee.Companion::descriptor")
```

Comment:
It'd be good to combine this with the code that generates a `FieldDescriptor` constructor in the `addFields()` method below. I.e. have one method that knows how to generate a `FieldDescriptor` constructor and call it from both places with appropriate arguments.